File: TestsManager/Service/Recording/RecordedEvents/AreaSelectEvent.cs
namespace Service.Recording.RecordedEvents
{
    public class AreaSelectEvent : IEvent
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public DateTime Timestamp { get; set; } = DateTime.Now;

        public required int Top { get; set; }
        public required int Bottom { get; set; }
        public required int Left { get; set; }
        public required int Right { get; set; }

        public required string AreaSelectViewFilepath { get; set; }

        public IEvent Clone()
        {
            return new AreaSelectEvent { Id = Id, Timestamp = Timestamp, Top = Top, Bottom = Bottom, Left = Left, Right = Right, AreaSelectViewFilepath = AreaSelectViewFilepath };
        }
    }
}

File: TestsManager/Frontend/src/app/models/events/area-select-event.ts
import { IEvent } from './event-interface';

export class AreaSelectEvent implements IEvent {
    public type: string = 'area-select';
    public id: string;
    public timestamp: Date;
    public top: number;
    public bottom: number;
    public left: number;
    public right: number;
    public areaSelectViewFilepath: string;
    public areaSelectViewBase64: string;

    constructor(
        id: string,
        timestamp: Date,
        top: number,
        bottom: number,
        left: number,
        right: number,
        areaSelectViewFilepath: string,
        areaSelectViewBase64: string
    ) {
        this.id = id;
        this.timestamp = timestamp;
        this.top = top;
        this.bottom = bottom;
        this.left = left;
        this.right = right;
        this.areaSelectViewFilepath = areaSelectViewFilepath;
        this.areaSelectViewBase64 = areaSelectViewBase64;
    }
}
